src/features/session_replay/aggregate/index.js

```diff
@@ -252,7 +252,7 @@ export class Aggregate extends AggregateBase {
     const { session } = getRuntime(this.agentIdentifier)
     if (!session.state.sessionReplaySentFirstChunk) this.syncWithSessionManager({ sessionReplaySentFirstChunk: true })
     this.recorder.clearBuffer()
-    if (recorderEvents.type === 'preloaded') this.scheduler.runHarvest()
+    if (recorderEvents.type === 'preloaded') this.scheduler.runHarvest(opts)
```

Comment:
It still seems inefficiently redundant to me that this runs unload harvest nested within the pagehide unload harvest callback, followed by vischange unload harvest callback so that prepareHarvest is being called 3 times here.

Reply:
I also thought we landed on sending 'preloaded' nodes immediately after feature-ready state in reg XHR instead.

Reply:
> It still seems inefficiently redundant to me that this runs unload harvest nested within the pagehide unload harvest callback, followed by vischange unload harvest callback so that prepareHarvest is being called 3 times here.

This is not happening, the problem that is happening is that unload is triggering prepareHarvest, and prepareHarvest is triggering other subsequent harvests (line 255) without the unload params so they are trying to be sent via xhr instead of beacon if that occurs.

Reply:
That's also as I understand it, but why is prepareHarvest calling into itself for 1, and 2, unloading is already being handled by the scheduler module itself, so while features can dictate modifications to payload if it's an unload, it's redundant to have features do the same job the scheduler does

Reply:
I'm not sure i'm following what you are proposing.  Can you dictate more clearly what strategy you are advocating for (specifically for handing preload payloads, of which there could be variable number of and of which could be excessively large due to snapshotting)?  There is currently no mechanism in the harvester to call for a feature to prep a payload multiple times.  which seems less of a duty for the harvest scheduler and more on the feature to manage that relationship.  There are a lot of places that our features trigger the scheduler to harvest immediately.  this just uses one more, and in doing so provides the rules that it did not use to.